Attempt to improve code analysis w/ Pair->record

labkey-jeckels opened the pull request:
#### Rationale
Hoping that using a more specific type for form and error data in BaseApiAction will improve code analysis results.

#### Changes
- Switch from using `Pair` to a dedicated `record` (BaseApiAction is the most interesting change here)
- Improve generics
- Delete dead code
- Stop calling deprecated methods

#### Tasks 📍
- Manual Testing - N/A
- Needs Automation - N/A

labkey-jeckels committed:
Attempt to improve code analysis w/ Pair->record

labkey-adam commented on query/src/org/labkey/query/controllers/QueryController.java:
```diff
@@ -3282,7 +3283,7 @@ public ActionURL getSuccessURL(QueryUpdateForm form)
                 str = form.getReturnUrl();
 
-            if (StringUtils.equals(str, "details.view"))
+            if (Strings.CS.equals(str, "details.view"))
```
Could be `"details.view".equals(str)`

labkey-adam commented on query/src/org/labkey/query/olap/BitSetQueryImpl.java:
```diff
@@ -1318,7 +1318,7 @@ Result union(OP op, Collection<Result> resultsIN) throws OlapException
                 throw new IllegalArgumentException();
             Hierarchy h = r.getHierarchy();
-            if (null == h || (null != hierarchyName && !StringUtils.equals(hierarchyName, h.getUniqueName())))
+            if (null == h || (null != hierarchyName && !Strings.CS.equals(hierarchyName, h.getUniqueName())))
```
Could be `!hierarchyName.equals(h.getUniqueName())` because null check

labkey-adam commented on study/src/org/labkey/study/model/DatasetDefinition.java:
```diff
@@ -632,7 +633,7 @@ public static TableInfo getTemplateTableInfo()
      * For consistency, now return the equivalent of
      *    StudyUserSchema().createSchema().getSchema("Datasets").getTable(_dataset.getLabel());
-     *
+     @*
```
Is this intentional?

labkey-adam commented on wiki/src/org/labkey/wiki/model/WikiVersion.java:
```diff
@@ -142,7 +142,7 @@ public HtmlString render(@Nullable HttpView<?> view, Container c, Wiki wiki)
     {
         if (null == _title)
-            _title = (null == _wikiName || StringUtils.equals(_wikiName, "default")) ? "Wiki" : _wikiName;
+            _title = (null == _wikiName || Strings.CS.equals(_wikiName, "default")) ? "Wiki" : _wikiName;
```
Could be `"default".equals(_wikiName)`

labkey-adam reviewed (approved):
See a few suggestions

Why the heck is every test run failing to bootstrap? I didn't see anything relevant in this PR.

labkey-jeckels committed:
Merge branch 'develop' into fb_improveCodeScanResults

labkey-jeckels committed:
Fix bootstrapping

labkey-jeckels commented:
> See a few suggestions
> 
> Why the heck is every test run failing to bootstrap? I didn't see anything relevant in this PR.

Thanks. I addressed those suggestions and found the cause of bootstrap failures. In DefaultModule I accidentally switched a `startsWith` to `endsWith`
